src/reflect/type.go

```diff
@@ -356,70 +379,45 @@ func TypeOf(i interface{}) Type {
 func PtrTo(t Type) Type { return PointerTo(t) }
 
 func PointerTo(t Type) Type {
-	if t.Kind() == Pointer {
+	switch t.Kind() {
+	case Pointer:
 		panic("reflect: cannot make **T type")
+	case Struct:
+		return (*structType)(unsafe.Pointer(t.(*rawType))).ptrTo
+	default:
+		return (*elemType)(unsafe.Pointer(t.(*rawType))).ptrTo
 	}
-	ptrType := t.(rawType)<<5 | 5 // 0b0101 == 5
-	if ptrType>>5 != t {
-		panic("reflect: PointerTo type does not fit")
-	}
-	return ptrType
 }
 
-func (t rawType) String() string {
+func (t *rawType) String() string {
 	return "T"
 }
 
-func (t rawType) Kind() Kind {
-	if t%2 == 0 {
-		// basic type
-		return Kind((t >> 1) % 32)
-	} else {
-		return Kind(t>>1)%8 + 19
-	}
+func (t *rawType) Kind() Kind {
+	return Kind(t.meta & 31)
 }
 
 // Elem returns the element type for channel, slice and array types, the
 // pointed-to value for pointer types, and the key type for map types.
-func (t rawType) Elem() Type {
+func (t *rawType) Elem() Type {
```

Comment:
The test corpus passing is probably a good indication most of this works, but I wonder if we could grab some tests from the upstream reflect package as a quick smoke test that's easier to run?

Reply:
So far I've used testdata/reflect.go as a test. That one covers most of the reflect package already.
Do you think we need to copy some tests from there?

Reply:
Good point, since this particular PR shouldn't be changing behaviour.  Checking that our existing tests continue to pass is probably enough.  We'll add more tests when we start adding more functionality.